### DISCUSSION TITLE
Ipopt statistic "t_wall_total" not available in C++

### DISCUSSION THREAD

**Discussion by salmagro** [casadi/casadi#3483] [Q&A] [answered]
Hello everyone,

I'm encountering an issue while trying to access the `t_wall_total` statistic from Ipopt. As shown in the screenshot below, this statistic seems to be available in the python API using:
![image](https://github.com/casadi/casadi/assets/52298571/97b805f8-53c0-4988-8c2d-6297494ddfa4)

This value matches the output of Ipopt when solving an optimization problem. 
![image](https://github.com/casadi/casadi/assets/52298571/fcee3682-c790-428b-9f43-bd917133faf2)

However, the same key is not available in the C++ map:
![image](https://github.com/casadi/casadi/assets/52298571/e76e7029-02d5-4579-840d-92ec7509daea)

Note that I am able to access statistics at the same level such as `iter_count`, `success`, or `iterations` in C++, but not `t_wall_total`. 

My understanding is that the CasADi Python API is essentially an abstraction of its implementation in C++. However, I can't find any relevant code line in version 3.4.x on GitHub.

I would greatly appreciate your insights and guidance on this matter. Or also another method to access the time that an optimization takes.

Thank you!

**Comment by victorfors** [comment]
You need to first call the stats() function (like you did in python) which gives you a dict where you can find solver specific statistics.

`casadi::Dict solve_stats = solver.stats();`

**Reply by salmagro** [reply]
Thanks for the reply, but I have already done it. Here is a more extended snip of my code:
```c++
        auto solver = getSolver().stats(); // Getter to get access to the solver

        msg.solver.iter_count = solver.at("iter_count").as_int(); // Available
	
	msg.solver.success = solver.at("success").as_bool(); // Available
	
	msg.solver.t_wall_total = solver.at("t_wall_total").as_double(); //  Not available
```

**Reply by victorfors** [reply]
Well, then it should work. Hard to help without a complete example.

Did you solve a problem with ipopt prior to calling .stats() using the same solver object as that obtained from getSolver()?

**Comment by jgillis** [comment, marked as answer]
Are you perchance setting `print_time` option to `false`? In that case, you'd need to make `record_time` true in order to get those timings.

**Reply by salmagro** [reply]
Indeed, the `record_time` was missing. Thank you very much @jgillis. Here is the documentation of that parameter for future reference: https://github.com/casadi/casadi/blob/a66c7b28d56a745aaa5c9b7e128139f9c1b99d4a/swig/doc.i#L11073-L11080
